ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -216,22 +216,33 @@ public void expireData(TableRuntime tableRuntime) {
       if (!expirationConfig.isValid(field, table.name())) {
         return;
       }
-      Instant startInstant;
-      if (expirationConfig.getSince() == DataExpirationConfig.Since.CURRENT_TIMESTAMP) {
-        startInstant = Instant.now().atZone(getDefaultZoneId(field)).toInstant();
-      } else {
-        startInstant =
-            Instant.ofEpochMilli(fetchLatestNonOptimizedSnapshotTime(table))
-                .atZone(getDefaultZoneId(field))
-                .toInstant();
-      }
 
-      expireDataFrom(expirationConfig, startInstant);
+      expireDataFrom(expirationConfig, expireBaseOnRule(expirationConfig, field));
     } catch (Throwable t) {
       LOG.error("Unexpected purge error for table {} ", tableRuntime.getTableIdentifier(), t);
     }
   }
 
+  protected Instant expireBaseOnRule(
+      DataExpirationConfig expirationConfig, Types.NestedField field) {
+    switch (expirationConfig.getBaseOnRule()) {
+      case CURRENT_TIME:
+        return Instant.now().atZone(getDefaultZoneId(field)).toInstant();
+      case LAST_COMMIT_TIME:
+        Snapshot snapshot = IcebergTableUtil.getSnapshot(getTable(), false);
+        long lastCommitTimestamp = fetchLatestNonOptimizedSnapshotTime(getTable());
+        // if the table does not exist any non-optimized snapshots, should skip the expiration
+        if (lastCommitTimestamp != Long.MAX_VALUE) {
+          // snapshot timestamp should be UTC
+          return Instant.ofEpochMilli(lastCommitTimestamp).atZone(ZoneOffset.UTC).toInstant();
+        } else {
+          return Instant.MIN;
+        }
+      default:
+        return Instant.MIN;
```

Comment:
Why is MIN? Maybe throw exception and catch it in outside?

Reply:
sure, will throw exception by default